app/controllers/landing_page_controller.rb

```diff
@@ -203,20 +220,48 @@ def render_landing_page(community_id:, default_locale:, locale_param:, structure
     )
 
     render_to_string :landing_page,
-           locals: { font_path: font_path,
+           locals: { font_path: FONT_PATH,
                      styles: landing_page_styles,
                      javascripts: {
-                       location_search: location_search_js
+                       location_search: location_search_js,
+                       translations: js_translations(locale_param, default_locale)
                      },
                      page: denormalizer.to_tree(structure, root: "page"),
-                     sections: denormalizer.to_tree(structure, root: "composition") }
+                     sections: denormalizer.to_tree(structure, root: "composition"),
+                     topbar_props: props,
+                     topbar_enabled: topbar_enabled }
   end
 
   def render_not_found(msg = "Not found")
     self.status = 404
     self.response_body = msg
   end
 
+  def topbar_props(community, community_customization, request_path, locale_param, default_locale)
+    # TopbarHelper pulls current lang from I18n
+    I18n.locale = locale_param.present? ? locale_param : default_locale
```

Comment:
`js_translations` method has this same logic. Would it make sense to move this logic a level up (to `render_landing_page` method) and name the result to `topbar_locale`. And then name the `locale` from structure to `landing_page_locale`. To me those names sound better and easier to understand than figuring out what does it mean if there is `locale_param` or `default_locale` or `locale`.